### PR TITLE
Add functionality for getting all/ recent transactions

### LINKED ISSUE #237: Add functionality for getting recent transactions

### PR DESCRIPTION
**Pull request by DimoDimchev**
- adds the frontend logic for getting recent transactions
- adds `/api/recent-transactions` endpoint to Portfolio
    - It accepts a `Recent` header. If the value is `True` only the last 6 transactions will be returned. Otherwise all transactions will be returned
    - each transaction has the following JSON structure:
```json
[
    "someId": {
        "amount": {
            "amount": "Float",
            "currency": "String",
        },
        "date": "yyyy-mm-dd",
        "type": "String"
    }
]
```
closes #237

### CODE REVIEW

**Review by KoliosterNikolayIliev — APPROVED** [review]

File: server/assets_v2/assets/blueprints/api.py [hunk 1]
```diff
@@ -68,4 +70,39 @@ async def get_transactions():
...
+    result = []
+    final = []
+    
+    try:
+        async with aiohttp.ClientSession() as session:
+            tasks = []
+            tasks.append(asyncio.ensure_future(get_all_yodlee_transactions(request.headers.get('yodlee_loginName'), session=session)))
+            await get_all_nordigen_transactions(requisitions=nordigen_requisitions, session=session, tasks=tasks)
+
+            responses = await asyncio.gather(*tasks)
```
Comment (Sheko1) [inline server/assets_v2/assets/blueprints/api.py:87]: you dont run transactions asynchronous

Comment (DimoDimchev) [reply server/assets_v2/assets/blueprints/api.py:87]: I do, the request for each account is added to tasks in the function in nordigen.py. The gather statement in api.py runs everything asynchronously

File: server/assets_v2/assets/utils/nordigen.py [hunk 2]
```diff
@@ -259,6 +256,39 @@ async def get_account_transactions(account, session):
...
+
+
+    for requisition in requisitions:
+        # get user bank accounts from requisition
+        accounts = await get_bank_accounts(requisition, session, headers)
+
+        # check requisition validation and check if user has bank accounts
+        if not accounts[1]:
+            # return error message
+            return accounts[0]
```
[the remark this answers is not in the document]
Comment (DimoDimchev) [reply server/assets_v2/assets/utils/nordigen.py:286]: We shouldn't have a requisition with no accounts, though, right?

[the remark this answers is not in the document]
Comment (Sheko1) [reply server/assets_v2/assets/utils/nordigen.py:286]: It is possible to have requisition with no accounts, example, user click the button to link his bank account with nordigen, but he does not finish the authentication then the user will have an empty requisition

File: server/assets_v2/assets/blueprints/api.py (hunk 1 above)
Comment (Sheko1) [reply server/assets_v2/assets/blueprints/api.py:87]: I see but you do requests to receive bank accounts from requisition wich is not in tasks

Comment (DimoDimchev) [reply server/assets_v2/assets/blueprints/api.py:87]: That part can't be made asynchronous. I would need to create a new session to run those requests asynchronously.

File: server/assets_v2/assets/utils/nordigen.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (DimoDimchev) [reply server/assets_v2/assets/utils/nordigen.py:286]: Okay, I will fix it

[the remark this answers is not in the document]
Comment (DimoDimchev) [reply server/assets_v2/assets/utils/nordigen.py]: Fixed in [4bb014c](https://github.com/DimoDimchev/Trivial/pull/240/commits/4bb041cc756629ec23e39a17b474ef153370f703)

**Comment by DimoDimchev** [comment]
Support for getting has been added but since that logic can't work asynchronously it slows down the process greatly. The reason why it can't work asynchronously is because we're using Coinbase's official Python library that uses `requests` under the hood. We could use direct `HTTP` requests using `aiohttp` instead but that's not possible because of the way we need to authenticate the user via the API. That sort of authentication is not supported by `aiohttp` yet

Sources:
[one](https://developers.coinbase.com/docs/wallet/api-key-authentication)
[two](https://github.com/aio-libs/aiohttp/issues/434)

File: server/assets_v2/assets/blueprints/api.py (hunk 1 above)
Comment (DimoDimchev) [reply server/assets_v2/assets/blueprints/api.py]: Fixed in [c05d7d9](https://github.com/DimoDimchev/Trivial/pull/240/commits/c05d7d9b29bebd4ff55fb4dd2b24ade9107852f2)

**Comment by DimoDimchev** [comment]
> Support for getting all Coinbase transactions has been added but since that logic can't work asynchronously it slows down the process greatly. The reason why it can't work asynchronously is because we're using Coinbase's official Python library that uses `requests` under the hood.
> 
> We could use direct `HTTP` requests using `aiohttp` instead but that's not possible because of the way we need to authenticate the user via the API. That sort of authentication is not supported by `aiohttp` yet.
> 
> At the current state the request for getting all of the transactions takes well over 10 seconds which is a big issue. A possible solution is moving to Coinbase's [OAuth authentication](https://developers.coinbase.com/docs/wallet/coinbase-connect/integrating) which eliminates the need for custom auth and would allow for the use of `aiohttp`.
> 
> @Sheko1 @KoliosterNikolayIliev @smanilov Please take a look at this.
> 
> Sources: [Coinbase's guide to authentication with API keys](https://developers.coinbase.com/docs/wallet/api-key-authentication) [A GitHub issue in `aiohttp`'s repository](https://github.com/aio-libs/aiohttp/issues/434)

Fixed in [31f9b0c](https://github.com/DimoDimchev/Trivial/pull/240/commits/31f9b0cfe4ce07e6f1433b048e349d4a3963326a)

**Review by KoliosterNikolayIliev — APPROVED** [review]

**Review by Sheko1 — APPROVED** [review]

**Pull request merged**